src/java.base/share/classes/sun/security/ssl/SSLTransport.java

```diff
@@ -122,16 +123,35 @@ static Plaintext decode(TransportContext context,
         } catch (AEADBadTagException bte) {
             throw context.fatal(Alert.BAD_RECORD_MAC, bte);
         } catch (BadPaddingException bpe) {
-            /*
-             * The basic SSLv3 record protection involves (optional)
-             * encryption for privacy, and an integrity check ensuring
-             * data origin authentication.  We do them both here, and
-             * throw a fatal alert if the integrity check fails.
-             */
-             Alert alert = (context.handshakeContext != null) ?
-                     Alert.HANDSHAKE_FAILURE :
-                     Alert.BAD_RECORD_MAC;
-            throw context.fatal(alert, bpe);
+            // Check for unexpected plaintext alert message during TLSv1.3 handshake, @bug 8331682
```

Comment:
Suggest to modify the comments here where re-constructing the plaintext instead of @bug 8331682.

Reply:
Will do, thanks!